internal/app/watch.go

```
@@ -0,0 +1,152 @@
+		refresh = m.commands.BindCmd(gen, refresh)
+	}
+	return m, tea.Batch(refresh, m.watch.tickCmd())
+}
+
+func (m Model) handleWatchRefresh(msg watchRefreshMsg) (tea.Model, tea.Cmd) {
+	if !m.watch.enabled || m.watch.target != msg.target || m.screen != msg.target {
+		return m, nil
+	}
+	m.watch.refreshing = true
+	updated, cmd := m.Update(msg.msg)
+	model := updated.(Model)
```

golangci-lint (2.12.2)

[error] type assertion must be checked

(forcetypeassert)